nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -755,3 +833,23 @@ def list_available_models(cls) -> Optional[PretrainedModelInfo]:
             )
         )
         return result
+
+    def on_save_checkpoint(self, checkpoint) -> None:
+        """LightningModule hook:
+        https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#on-save-checkpoint
+        """
+        if isinstance(self.model, list):
+            for i in range(len(self.model)):
+                parallel_state.set_virtual_pipeline_model_parallel_rank(i)
+                checkpoint[f'model{i}'] = self.model[i].module.state_dict_for_save_checkpoint()
+            parallel_state.set_virtual_pipeline_model_parallel_rank(0)
```

Comment:
so the checkpoint file is not compatible with our old non-interleaved checkpoint files?

Reply:
I don't know what we should do about this. It seems to make them compatible we need to consider all of our models as a list (old models will be list of length 1). Is there a way around this?